roofit/roofitcore/inc/RooDataHist.h

```diff
@@ -279,7 +279,7 @@ class RooDataHist : public RooAbsData, public RooDirItem {
   mutable std::vector<double> _maskedWeights; //! Copy of _wgt, but masked events have a weight of zero.
   mutable std::vector<double> _maskedSumw2; //! Copy of _sumW2, but masked events have a weight of zero.
  
-  mutable unsigned long _curIndex{std::numeric_limits<std::size_t>::max()}; // Current index
+  mutable unsigned long _curIndex{static_cast<unsigned long>(std::numeric_limits<std::size_t>::max())}; // Current index
```

Comment:
Can you not return here:
std::numeric_limits<unsigned long>::max() ? 

But what is unsigned_long and size_t on Win64? 
Maybe a better alternative would be to make the function returning a size_t instead of an unsigned long.

Reply:
See [this commit](https://github.com/root-project/root/commit/8c161202948e2ffa4d2ffc3e2d8f4b20f72aad9e) who broke the build on Win64

Reply:
Hi @lmoneta, this is not a function but a member variable, which can't be `size_t` because then the schema is not platform independent.

This still means we should go with `std::numeric_limits<unsigned long>::max()` here, because that's what we want to initialize `_curIndex` with: the maximum value it can hold.

Thanks @bellenot for this PR by the way!

Reply:
@guitargeek `unsigned long` is not platform independent. e.g. it is 32 bit on Win64

Reply:
If you really want a platform independent value then you should use here `ULong64_t` or `uint64_t`.

Reply:
@moneta @guitargeek Do it the way you prefer, but this should be fixed somehow...

Reply:
Do you actually need to save `_curIndex`? If so, please use a `Long64_t` as member type and as the template arg to `numeric_limits`. If not, please mark it as transient with `//!` and use `numeric_limits<unsigned long>`.